File: split_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def split_file(file_path, lines_per_file, prefix, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        lines = f.readlines()
    for i in range(0, len(lines), lines_per_file):
        output_path = file_path.parent / f"{prefix}{i//lines_per_file}.txt"
        with open(output_path, "w", encoding="utf-8") as f:
            f.writelines(lines[i:i+lines_per_file])
    return f"Split {file_path} into {len(lines)//lines_per_file + 1} files"

File: ts-tools/split_tool.ts
import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function splitFile(
  filePath: string,
  linesPerFile: number,
  prefix: string,
  config: any,
): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'read', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.readFile(resolvedPath, 'utf-8', (err, data) => {
          if (err) {
            reject(err);
          } else {
            const lines = data.split('\n');
            let fileCount = 0;
            for (let i = 0; i < lines.length; i += linesPerFile) {
              const outputPath = path.join(
                path.dirname(resolvedPath),
                `${prefix}${fileCount}.txt`,
              );
              const chunk = lines.slice(i, i + linesPerFile);
              fs.writeFile(outputPath, chunk.join('\n'), 'utf-8', (err) => {
                if (err) {
                  reject(err);
                }
              });
              fileCount++;
            }
            resolve(`Split ${filePath} into ${fileCount} files`);
          }
        });
      })
      .catch(reject);
  });
}
